packages/dependency-extraction-webpack-plugin/lib/util.js

```diff
@@ -56,6 +56,21 @@ function defaultRequestToExternal( request ) {
 	}
 }
 
+/**
+ * Default request to external module transformation
+ *
+ * Currently only @wordpress/interactivity
+ *
+ * @param {string} request Module request (the module name in `import from`) to be transformed
+ * @return {string|undefined} The resulting external definition. Return `undefined`
+ *   to ignore the request. Return `string` to map the request to an external. This may simply be returning the request, e.g. `@wordpress/interactivity` maps to the external `@wordpress/interactivity`.
+ */
+function defaultRequestToExternalModule( request ) {
+	if ( request === '@wordpress/interactivity' ) {
+		return request;
```

Comment:
If here I return something else than `request`, for example, a different (mapped) module name, will the plugin transform the `import '@wordpress/interactivity'` to `import '@wordpress/mapped-interactivity'`?

And will it output the _mapped_ name to the `.asset.php` file? I suspect it will output the original one.

The "scripts" version has the `requestToHandle` function to produce the handles that go into the `.asset.php` file.

Reply:
You're right, that's a bug. If we map to a different name, include the original in the asset file. I'll address it and add a test.

I don't think we need both functions for modules (toExternal and toHandle), we should be able to use a single module ID for the external module import and the WordPress module ID.

I'd even consider changing the signature (and name of the option) so that it expects to return `true` to indicate we want to externalize the module and track the dependency. It doesn't seem like we need to be rewriting the imports.

Any thoughts on that?

Reply:
After some more reflection, there are likely use cases for rewriting module names.

For example, maybe I want to include a specific version of a package on NPM, but I'd like to rename it in the WordPress module system to avoid collisions. I can use the regular import with the npm name in my source files, but rewrite to a unique name for the external and the WordPress module ID.

I still don't see the need for two functions (toExternal and toHandle), one seems sufficient.

Reply:
> I still don't see the need for two functions (toExternal and toHandle), one seems sufficient.

That's fine, I agree that for modules we need only one. There are only module names (like `@wordpress/interactivity`), nothing else. With `scripts` there are two distinct entities: the `window.wp.blob` global variable produced by `toExternal` and the `wp-blob` WP script handle produced by `toHandle`.

Taking care to add the mapped value produced by `toExternal` to `.asset.php`, not the original one, is enough to address my concerns.

> After some more reflection, there are likely use cases for rewriting module names.

There could be also a `true` return value that says "externalize without changing the name". Equivalent to returning the `request` value.

Reply:
I pushed a test and a fix for the renamed module bug. Thanks! I don't plan to implement the `true` shorthand to use the same module ID for now, but it's easy enough if we'd like to add it at some point.

Reply:
> I don't plan to implement the `true` shorthand to use the same module ID for now

I think it's a good idea to have this shorthand. The evidence is that literally all use cases we currently have, both in production and it tests, could use the `true` return value. Another thing to add to my todo 🙂

Reply:
https://github.com/WordPress/gutenberg/pull/57593